_rules/form-field-visual-label-is-descriptive-1cd4da.md

```diff
@@ -0,0 +1,211 @@
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.4.6: Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html)
+- [G131: Providing descriptive labels](https://www.w3.org/WAI/WCAG21/Techniques/general/G131)
+- [H44: Using label elements to associate text labels with form controls](https://www.w3.org/WAI/WCAG21/Techniques/html/H44)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [visible][] `label` element describes the purpose of the associated element.
+
+```html
+<label for="fname">First name:</label> <input id="fname" type="text" name="fname" />
+```
+
+#### Passed Example 2
+
+The implicit label, that is coded with the `label` element, describes the purpose of the associated element.
+
+```html
+<label>First name:<input id="fname" type="text" name="fname"/></label>
+```
+
+#### Passed Example 3
+
+The headings provide a [visual context][] that differentiates the purpose of the otherwise identically named form fields, [visual context][] does not need to be [included in the accessibility tree][].
+
+```html
+<h2 aria-hidden="true">Shipping address</h2>
+<label>Name<input id="shipping-name" type="text" name="name"/></label>
+<label>Street<input id="shipping-street" type="text" name="street"/></label>
+
+<h2 aria-hidden="true">Billing address</h2>
+<label>Name<input id="billing-name" type="text" name="name"/></label>
+<label>Street<input id="billing-street" type="text" name="street"/></label>
+```
+
+#### Passed Example 4
+
+The [visible][] `label` element describes the form field.
+
+```html
+<label for="fname">First name:</label> <input id="fname" type="text" name="fname" aria-label="Fill in:" />
+```
+
+#### Passed Example 5
+
+The [visible][] `label` element describes the form field.
+
+```html
+<label for="fname">First name:</label> <input id="fname" type="text" name="fname" aria-label="" />
+```
+
+### Failed
+
+#### Failed Example 1
+
+The label that is coded with the `label` element does not describe the purpose of the associated element.
+
+```html
+<label for="fname">Menu</label> <input id="fname" type="text" name="fname" />
+```
+
+#### Failed Example 2
+
+The implicit label that is coded with the `label` element does not describe the purpose of the associated element.
+
+```html
+<label>Menu<input id="fname" type="text" name="fname"/></label>
+```
+
+#### Failed Example 3
+
+The `label` in itself does not describe the purpose of the form field. While the [context][] can be relied upon to differentiate form fields from each other, it cannot be relied upon for describing the entire purpose of the form field.
+
+```html
+<h2>Name</h2>
+<label>Fill in: <input id="name" type="text" name="name"/></label>
+```
```

Comment:
Per the rule, this TC is a failed one. But regarding the SC, I don't think this is a failed, since h2 can be taken as the visible label (even if not marked as). Isn't it a passed on for the SC?

Reply:
Yes. 😭